include/dynd/parse_util.hpp

```diff
@@ -1056,8 +1056,9 @@ std::enable_if_t<is_floating_point<T>::value, T> parse(const char *begin, const
 
   // TODO: use http://www.netlib.org/fp/dtoa.c
   char *end_ptr;
-  T value = strto<T>(begin, &end_ptr);
-  if (end_ptr - begin != end - begin) {
+  const char *cp = std::string(begin, end).c_str();
```

Comment:
We could use a buffer on the stack for regularly sized inputs and fall back to std::string or similar for huge inputs.  I think the memcpy() into a static buffer is probably negligible compared to strtod() and for sure compared to dtoa.c (mentioned in the TODO comment).